app/Tools/File.php

```diff
@@ -191,37 +193,7 @@ public function addToJson(string $path, string $key, string|array $value): stati
     // @todo: Improve this to handle traits, interfaces, etc.
     public function addImport(string $path, string $class): static
     {
-        $useString = "use $class;\n";
-
-        if (str_contains($contents = Storage::get($path), $useString)) {
-            return $this;
-        }
-
-        $contents = explode("\n", $contents);
-
-        // Find the first line that starts with `class`
-        $classIndex = array_search(
-            'class',
-            array_map(fn ($line) => str_starts_with($line, 'class') ? substr($line, 0, 5) : null, $contents)
-        );
-
-        if ($classIndex === false) {
-            throw new Exception("Class keyword not found in {$path}");
-        }
-
-        // Insert the use statement just before the class definition
-        $newContents = array_merge(
-            array_slice($contents, 0, $classIndex),
-            [$useString],
-            array_slice($contents, $classIndex)
-        );
-
-        if (! $newContents[$classIndex - 1] && str_starts_with($newContents[$classIndex - 2], 'use')) {
-            // Remove line above this import entirely... but only if this is the only import
-            unset($newContents[$classIndex - 1]);
-        }
-
-        Storage::put($path, implode("\n", $newContents));
+        (new Editor())->edit($path, [new AddImportVisitor($class)]);
```

Comment:
I like the idea of the `File` class having the nice syntax, and instead of holding its own implementation details, it holds knowledge of how to configure and use the `Editor` class to have it handle implementation details.